src/web/content.rs

```diff
@@ -42,50 +38,55 @@ const PAGINATION_LIMIT_MAX: usize = 20;
 #[derive(Debug)]
 pub struct Input<T>(pub T);
 
-impl<'a, T> FromData<'a> for Input<T>
+#[rocket::async_trait]
+impl<'r, T> FromData<'r> for Input<T>
 where
-    T: Deserialize<'a>,
+    T: DeserializeOwned,
 {
     type Error = ApiError;
-    type Owned = String;
-    type Borrowed = str;
 
-    fn transform(r: &Request, d: Data) -> Transform<Outcome<Self::Owned, Self::Error>> {
-        let size_limit = r
+    async fn from_data(request: &'r Request<'_>, data: Data<'r>) -> data::Outcome<'r, Self> {
+        let size_limit = request
             .limits()
             .get("max_request_size")
-            .unwrap_or(MAX_REQUEST_SIZE);
-
-        let mut buf = String::with_capacity(8192);
-        match d.open().take(size_limit).read_to_string(&mut buf) {
-            Ok(_) => Borrowed(Success(buf)),
-            Err(e) => Borrowed(Failure((
-                Status::BadRequest,
-                ApiError::BadRequest(e.to_string()),
-            ))),
-        }
-    }
+            .unwrap_or_else(|| MAX_REQUEST_SIZE.bytes());
```

Comment:
Why `.unwrap_or_else()` instead of `.unwrap_or()`?

Reply:
I believe it's a `clippy` suggestion. For anything other than a constant, it prefers a closure, so that it can be evaluated lazily.